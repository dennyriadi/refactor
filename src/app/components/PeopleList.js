import R from 'ramda';
import React from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v4';

const PeopleList = ({ people }) => {
  const styleByGender = person => person.gender === 'male' ? 'info' : 'warning';

  return (
    <div className="container-fluid">
      <ul className="list-group">
        {
          R.map(person =>
            <li
              key={uuid()}
              className={`list-group-item list-group-item-${styleByGender(person)}`}>
              {`${person.name} (${person.age})`}
            </li>
          )(people)
        }
      </ul>
    </div>
  );
};

PeopleList.propTypes = {
  people: PropTypes.array.isRequired,
};

export default PeopleList;