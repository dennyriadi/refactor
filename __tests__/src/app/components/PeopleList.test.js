import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import PeopleList from '../../../../src/app/components/PeopleList';

describe('<PeopleList />', () => {
  describe('render()', () => {
    test('should render without people list', () => {
      const wrapper = shallow(<PeopleList people={[]} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    test('should render correctly', () => {
      const people = [
        { name: 'John', age: 30, gender: 'male' },
      ];
      const wrapper = shallow(<PeopleList people={people} />);
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
