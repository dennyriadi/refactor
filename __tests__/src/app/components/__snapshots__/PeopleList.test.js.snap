// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PeopleList /> render() should render correctly 1`] = `
<div
  className="container-fluid"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item list-group-item-info"
    >
      John (30)
    </li>
  </ul>
</div>
`;

exports[`<PeopleList /> render() should render without people list 1`] = `
<div
  className="container-fluid"
>
  <ul
    className="list-group"
  />
</div>
`;
